server/service/src/invoice/inbound_shipment/batch.rs

```diff
@@ -177,11 +177,21 @@ pub fn batch_inbound_shipment(
                 return Err(WithDBError::err(results));
             }
 
-            let (has_errors, result) = mutations_processor
-                .do_mutations_with_user_id(input.update_shipment, update_inbound_shipment);
-            results.update_shipment = result;
-            if has_errors && !continue_on_error {
-                return Err(WithDBError::err(results));
+            if let Some(update_shipment_inputs) = input.update_shipment {
+                let output: Vec<_> = update_shipment_inputs
+                    .into_iter()
+                    .filter_map(|input| {
+                        let result = update_inbound_shipment(ctx, input.clone(), None);
```

Comment:
All these changes are because I changed the `update_inbound_shipment` fn signature with the new 3rd parameter, so it no longer fit the signature `do_mutations_with_user_id` which expects the callback to have 2 parameters

Reply:
It's worth noting that functionally it is slightly different, but I suspect that the old generic method is doing it a less than ideal way:

https://github.com/msupply-foundation/open-msupply/blob/73029fd38a3e4076dddcd59d08f54fc7a3096456/server/service/src/lib.rs#L228-L250

After the first error is encountered, it'll include all subsequent results even if they don't contain errors. The implementation I have done here for `update_inbound_shipment` only includes results that contain errors.